backend/controller/dal/internal/sql/queries.sql

```diff
@@ -123,8 +123,8 @@ ORDER BY r.key;
 SELECT sqlc.embed(d), m.name AS module_name, m.language, COUNT(r.id) AS replicas
 FROM deployments d
   JOIN modules m ON d.module_id = m.id
-  LEFT JOIN runners r ON d.id = r.deployment_id
-WHERE min_replicas > 0 AND r.state = 'assigned'
+  LEFT JOIN runners r ON d.id = r.deployment_id AND r.state = 'assigned'
```

Comment:
It changes the query behaviour. If you have an 'idle' runner with the current query it will join, the deployment, then get filtered out in the where clause. Filtering it here means that it just won't join, so the runner count will be zero but you still see the deployment row.